### PR TITLE
Maintain and typo fix

### PR DESCRIPTION
**Pull request by Zonaldo7**
1. Remove with statement when constructing pulses, compatible with Qiskit 1.1.1
2. Fix the error of multiprocessing when retrying to get job results on Mac and Windows
3. Fix typos in notebook2

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
+qiskit-dynamics == 0.5.1
+qiskit-experiments == 0.7.0
+qiskit-ibm-runtime == 0.24.1
+qiskit-ibm-experiment == 0.4.7
 uncertainties
 lmfit
 pylatexenc
 qutip == 5.0.*
-jax[cpu] == 0.4.23
+jax[cpu] == 0.4.6
```
Comment (BoxiLi) [inline requirements.txt]: Why is jax downgraded to `0.4.6`?

In fact, even `jax[cpu]==0.4.23` has a conflict with  `scipy>1.12` I think. I got the error.


```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Local/bli/qiskit/qiskit-CR-calibration/qiskit_utilities/__init__.py", line 5, in <module>
    from .fake_backend import *
  File "/Local/bli/qiskit/qiskit-CR-calibration/qiskit_utilities/fake_backend.py", line 10, in <module>
    from qiskit_dynamics.array import Array
  File "/local/bli/conda/qiskit1/lib/python3.11/site-packages/qiskit_dynamics/__init__.py", line 38, in <module>
    from .solvers.solver_functions import solve_ode, solve_lmde
  File "/local/bli/conda/qiskit1/lib/python3.11/site-packages/qiskit_dynamics/solvers/__init__.py", line 110, in <module>
    from .solver_functions import solve_ode, solve_lmde
  File "/local/bli/conda/qiskit1/lib/python3.11/site-packages/qiskit_dynamics/solvers/solver_functions.py", line 39, in <module>
    from .fixed_step_solvers import (
  File "/local/bli/conda/qiskit1/lib/python3.11/site-packages/qiskit_dynamics/solvers/fixed_step_solvers.py", line 34, in <module>
    from jax.scipy.linalg import expm as jexpm
  File "/local/bli/conda/qiskit1/lib/python3.11/site-packages/jax/scipy/linalg.py", line 18, in <module>
    from jax._src.scipy.linalg import (
  File "/local/bli/conda/qiskit1/lib/python3.11/site-packages/jax/_src/scipy/linalg.py", line 409, in <module>
    @_wraps(scipy.linalg.tril)
            ^^^^^^^^^^^^^^^^^
AttributeError: module 'scipy.linalg' has no attribute 'tril'
```

File: qiskit_utilities/fake_backend.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 from qiskit_dynamics.array import Array
 
 Array.set_default_backend("jax")
-from qiskit.test.mock import FakeLagos
+from qiskit_ibm_runtime.fake_provider import FakeLagos
```
Comment (BoxiLi) [inline qiskit_utilities/fake_backend.py:13]: The inclusion of this was a mistake I think, it is not used anywhere. Can remove.

**Review by BoxiLi — COMMENTED** [review]
Thanks a lot! @Zonaldo7 
What is changed in the notebook 2? Just because the tracking does not work well with ipynb.

**Comment by Zonaldo7** [comment]
In the notebook2, retrieve_expdata is spelled as retrive_expdata. The downgrade of jax is a mistake, which should be upgraded to 0.4.25 instead. Also, as the perturbation module is not used throughout the project, using a jax version > 0.4.6 works fine for me.

**Comment by BoxiLi** [comment]
Ok! `0.4.25` works for me.

File: requirements.txt (hunk 1 above)
Comment (BoxiLi) [inline requirements.txt]:
```suggestion
jax[cpu] == 0.4.25
```

**Pull request merged**